docs: typos & other grammar updates to moderator-handbook (#47039)
freeCodeCamp/freeCodeCamp: docs/moderator-handbook.md

# The Official freeCodeCamp Moderator Handbook

This handbook will help you moderate different places in our community. This covers conversations and interactions in issues and pull request threads on GitHub, the community forum, the chat rooms, and other official communities that we foster.

> [!NOTE]
> All freeCodeCamp moderators are community-wide moderators. That means we trust you to oversee any of these places.

You can serve as a moderator on any of the platforms that are of the most interest to you. Some moderators just help out on GitHub, while others just help out on the forum. Some moderators are active everywhere.

> The bottom line is that we want you to enjoy being a moderator, and invest your free time in places that are of interest to you.

Reviewer: ```suggestion
The bottom line is that we want you to enjoy being a moderator, and invest your scarce time in places that are of interest to you.
```

it's not free time, it's time being invested in being moderators

> "With great power comes great responsibility." - Uncle Ben

As a moderator, temperament is more important than technical skill.

Listen. Be helpful. Don't abuse your power.

> FreeCodeCamp is an inclusive community, and we need to keep it that way.

Reviewer: ```suggestion
freeCodeCamp is an inclusive community, and we need to keep it that way.
```

it's a registered name, it should be written like this

## Moderating GitHub

Moderators have two primary responsibilities on GitHub:

1. Triaging and responding to issues.
2. Reviewing and merging pull requests (aka QA).

### Moderating GitHub issues

We use our main [`freeCodeCamp/freeCodeCamp`](https://github.com/freeCodeCamp/freeCodeCamp/issues) repository as a common issue tracker for all of our repositories. We get new issues every day, all of which need to be triaged, labeled, and addressed. This is also a great place to start helping with open-source codebase contributions.

#### Issue triage

[Triaging](https://en.wikipedia.org/wiki/Triage) is a process of prioritizing attention to each new issue report. We have an extensive list of labels that we use to mark each issue's priority, category, status, and scope.

You can help us organize and triage the issue reports by applying labels from [this list](https://github.com/freeCodeCamp/freeCodeCamp/labels). Usually, a description is available alongside the label explaining its meaning.

> Please pay special attention to the labels `"help wanted"` and `"first-timers only"`. These are to be added to threads that you think can be opened up to potential contributors for making a pull request.

Reviewer: ```suggestion
Please pay special attention to the labels `"help wanted"` and `"first timers only"`. These are to be added to threads that you think can be opened up to potential contributors for making a pull request.
```

the label is "first timers only"

Author: For this, wouldn't the labels have to be updated to fit to "first-timers only"? I plan on reading through the rest of freeCodeCamp's documentation to fix any inconsistent punctuation with a dictionary (first-timer is a compound noun).

Author: Talked with Naomi about any labels regarding the words first-time and first-timer, they will be left unhyphenated.

> A `"first-timer only"` label should be applied to a trivial issue (ie a typo fix) and should include additional information. You can use this [reply template](moderator-handbook.md#first-timer-only-issues) for triage.

Reviewer: ```suggestion
A `"first timer only"` label should be applied to a trivial issue (ie a typo fix) and should include additional information. You can use this [reply template](moderator-handbook.md#first-timer-only-issues) for triage.
```

> ##### Handling merge conflicts:

Reviewer 2: ```suggestion
##### Handling Merge Conflicts:
```

These are opportunities for you to welcome them and make them feel extra special. You never know which person who's marginally involved may become our next super-helper, helping many other people in their coding journey. Even the slightest kindness may trigger a cascade of good deeds.

### Deleting Forum Posts

Forum moderators can delete users' posts. You should only do this for the following instances:

1. Someone has posted a pornographic or graphically violent image.
2. Someone has posted a link or code that is malicious in nature and could harm other campers who click on it.
3. Someone has flooded a thread with a lot of spam messages.

### Dealing with Spam

For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin panel). Send the user a warning with the [Code of Conduct](https://code-of-conduct.freecodecamp.org). Check the box in the private message indicating that your message is a "formal warning."

> As a moderator, you can ask questions and report incidents in the [staff forum section](https://forum.freecodecamp.org/c/mod-team/4).

Reviewer 2: ```suggestion
As a moderator, you can ask questions and report incidents in the [mod-team forum section](https://forum.freecodecamp.org/c/mod-team/4).
```

@ieahleen's comment.

> ##### Closing an invalid pull request:

Reviewer 2: ```suggestion
##### Closing an Invalid Pull Request:
```

Often, a pull request will be low effort. You can usually tell this immediately when the contributor didn't bother checking the checkboxes in the Pull Request Template or used a generic pull request title like "Made changes" or "Update index.md".

There are also situations where the contributor is trying to add a link to their website, include a library they created, or have a frivolous edit that doesn't help anyone but themselves.

You can close these invalid pull requests and reply to them with these [reply templates](moderator-handbook.md#closing-invalid-pull-requests).

> #### Other guidelines for moderators on GitHub

Reviewer 2: ```suggestion
#### Other Guidelines for Moderators on GitHub
```

Though you will have write access to freeCodeCamp's repository, **you should never push code directly to freeCodeCamp repositories**. All code should enter freeCodeCamp's codebase in the form of a pull request from a fork of the repository.

Also, you should never accept your own PRs. They must be reviewed by another moderator, just like any other PR.

If you notice anyone breaking the [Code of Conduct](https://code-of-conduct.freecodecamp.org) on GitHub issues, or opening pull requests with malicious content or code, email `support[at]freecodecamp.org` with a link to the offending pull request, and we can consider banning them from freeCodeCamp's GitHub organization entirely.

> ## Moderating the forum

Reviewer 2: ```suggestion
## Moderating the Forum
```

> ### Deleting forum posts

Reviewer 2: ```suggestion
### Deleting Forum Posts
```

Forum moderators can delete users' posts. You should only do this for the following instances:

1. Someone has posted a pornographic or graphically violent image.
2. Someone has posted a link or code that is malicious in nature and could harm other campers who click on it.
3. Someone has flooded a thread with a lot of spam messages.

> ### Dealing with spam

Reviewer 2: ```suggestion
### Dealing With Spam
```

For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin panel). Send the user a warning with the [Code of Conduct](https://code-of-conduct.freecodecamp.org). Check the box in the private message indicating that your message is a "formal warning."

As a moderator, you can ask questions and report incidents in the [staff forum section](https://forum.freecodecamp.org/c/mod-team/4).

> ### Dealing with off-topic conversations

Reviewer 2: ```suggestion
### Dealing With Off-Topic Conversations
```

Posts or topics that seem to be in the wrong place can be recategorized or renamed to whatever would be appropriate.

In exceptional circumstances, it may be appropriate for a moderator to fork a discussion into multiple threads.

> Again, if you have any problems or questions, make a post with your actions in the staff category, and tag another moderator if you want them to review your moderating actions.

Reviewer: ```suggestion
Again, if you have any problems or questions, make a post with your actions in the Staff category, and tag another moderator if you want them to review your moderating actions.
```

the category is "Staff"

> ### Underage users

Reviewer 2: ```suggestion
### Underage Users
```

But if it is an egregious offense that can't reasonably be attributed to a cultural difference or a misunderstanding of the English language. In that case, you should strongly consider blocking the member from the Facebook group.

> ## Moderating chat

Reviewer 2: ```suggestion
## Moderating Chat
```

   Evidence: _One or more links to the offending message(s)_
   ```

3. **Creating a private discussion**

   There may be situations where you need to address a concern with a camper privately. This should not be done through DMs, which can lead to situations where you claim one thing and the camper claims another. Instead, use the bot's functionality to create a private discussion:

   - Call the `!fcc private username` command, where `username` is the camper's chat username.
   - The bot will create a new channel, and add the mentioned camper and all moderators with the `Your Friendly Moderator` role. While all moderators are added to the channel for transparency, the moderator who calls this command should be the only one to interact with the camper unless they request assistance.

>    - When the conversation is complete, call the `!fcc close` command _in the private channel_ to have the bot close and delete that channel.

Reviewer: Note: this needs to be updated
cc/ @naomi-lgbt

Reviewer 2: ```suggestion
   - When the conversation is complete, click the `❌ Close` button _on the first message in the private channel_ to have the bot close and delete that channel.
```

   - For mild and unintentional violations of the [Code of Conduct](https://code-of-conduct.freecodecamp.org):

     Moderator: This is a friendly reminder for everyone to follow the [Code of Conduct](https://code-of-conduct.freecodecamp.org): https://code-of-conduct.freecodecamp.org/

7. **Don’t brag about being a moderator**

   Do not see yourself as above the community. **You are the community.** And the community has trusted you to help protect something rare that we all share - a _welcoming_ place for new developers.

   If you brag about being a moderator, people may feel uneasy around you, in the same way that people may feel uneasy around a police officer, even if they’re doing nothing wrong. This is just human nature.

8. **Don’t contradict other moderators**

   If you disagree with a moderator's action, talk with them in private or bring it up in the #mod-chat channel. Never override a moderator's action, and never contradict the other moderator(s) publicly. Instead, have a cool-headed discussion in `#mod-chat` and convince the moderator that they themselves should reverse their ban or change their PoV (Point of View).

>    *Remember: We’re all on the same team. We want to dignify the role of moderators and present a unified front.*

Reviewer 2: ```suggestion
   _Remember: We’re all on the same team. We want to dignify the role of moderators and present a unified front._
```

Think this was my fault. Prettier expects underscores for italics instead of asterisks. 

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

   > [!NOTE]
   > If the message contains material that would be illegal to take a screenshot of, copy the message link instead - provide that message link to **@raisedadead** to forward to Discord's Trust and Safety team.

5. **Don’t use @all, @everyone, or @here**

   Don’t use **@all**, **@everyone**, or **@here** under any circumstances! Every single person in that chat room will get a notification. In some cases, tens of thousands of people.

   Instead, if you want people to see an announcement, you can pin it to the channel to allow everyone to read it over time.

6. **Don’t threaten to take action**

   If a camper breaks the [Code of Conduct](https://code-of-conduct.freecodecamp.org), don’t threaten to take moderator action, and never warn them in public. Instead, talk to them privately using the bot's `private` command. No one else in that channel needs to know that you banned/suspended the person. If a violation was clearly unintended and doesn't warrant a suspension or private conversation, make the offending camper aware of his / her actions without making it come across as a warning.

>    

Reviewer 2: ```suggestion

```

Empty blankspace in this line needs to be removed.

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

   - For mild and unintentional violations of the [Code of Conduct](https://code-of-conduct.freecodecamp.org):

     Moderator: This is a friendly reminder for everyone to follow the [Code of Conduct](https://code-of-conduct.freecodecamp.org): https://code-of-conduct.freecodecamp.org/

7. **Don’t brag about being a moderator**

   Do not see yourself as above the community. **You are the community.** And the community has trusted you to help protect something rare that we all share - a _welcoming_ place for new developers.

   If you brag about being a moderator, people may feel uneasy around you, in the same way that people may feel uneasy around a police officer, even if they’re doing nothing wrong. This is just human nature.

8. **Don’t contradict other moderators**

   If you disagree with a moderator's action, talk with them in private or bring it up in the #mod-chat channel. Never override a moderator's action, and never contradict the other moderator(s) publicly. Instead, have a cool-headed discussion in `#mod-chat` and convince the moderator that they themselves should reverse their ban or change their PoV (Point of View).

>    *Remember: We’re all on the same team. We want to dignify the role of moderators and present a unified front.**

Reviewer 2: ```suggestion
   *Remember: We’re all on the same team. We want to dignify the role of moderators and present a unified front.*
```

This should fix the prettier issue.

__
[![sema-logo](https://app.semasoftware.com/img/sema-tray-logo.gif)](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

9. **Talk with other moderators**

   We have a room for moderators only. Use it! If you feel uncomfortable with handling a certain situation, ask other moderators for help. If you think something should be discussed, do it. You're part of the team, and we value every team member's input! Even if you totally disagree with anything in these guidelines or the [Code of Conduct](https://code-of-conduct.freecodecamp.org)!

10. **Temporarily inactive**

    If you're not going to be active as a Moderator for a while due to vacation, illness, or any other reason, make sure to let the others know in the `#mod-chat` channel. This is so we know if we can count on you to be regularly active on the server or not.

> ## How to become a moderator

Reviewer 2: ```suggestion
## How to Become a Moderator
```

Suppose you are helping people in the community consistently over time. In that case, our moderator team will eventually take notice, and one of them will mention you as a possible moderator to [our staff](https://forum.freecodecamp.org/g/Team). There are no shortcuts to becoming a moderator.

If you are approved, we will add you to our moderator teams on [GitHub](https://github.com/orgs/freeCodeCamp/teams/moderators), [forum](https://forum.freecodecamp.org/g/moderators), chat, etc.

> [!NOTE]
> For GitHub: After you've been accepted as a moderator, you will receive a Github repository invitation. You'll need to head over towards [freeCodeCamp GitHub Organization Invitation](https://github.com/orgs/freeCodeCamp/invitation) to be able to accept the invitation.
>
> This is required for us to be able to give you write access to some of our repositories.

> ## How we retire inactive moderators

Reviewer 2: ```suggestion
## How We Retire Inactive Moderators
```

Please note that we will frequently remove moderators whom we think are inactive. When we do this, we will send the following message:

```markdown
This is a standard message notifying you that, since you don't seem to have been an active moderator recently, we're removing you from our moderator team. We deeply appreciate your help in the past.

If you think we did this in error, or once you're ready to come back and contribute more, just reply to this message letting us know.
```

> ## How our contributors room works

Reviewer 2: ```suggestion
## How Our Contributors Room Works
```

Anyone is welcome in the [contributors room on our chat server](https://discord.gg/PRyKn3Vbay). It is the designated chat room for moderators and other campers who contribute to our community in any number of ways, including through study groups.

We assume contributors will read anything in this room that directly mentions them with an **@username**. Everything else is optional, but feel free to read anything anyone posts in there and interact.

> ## Dealing with solicitors

Reviewer 2: ```suggestion
## Dealing With Solicitors
```

As a matter of policy, you should avoid talking privately about these matters. Should the situation reflect back to freeCodeCamp, we want to have the conversation(s) on record. Make it clear that we are not medical professionals and that you encourage the user to find professional help.

As difficult as it sometimes can be, avoid giving any tips or advice and rather point the user in the direction of seeking professional help!

If this happens on our chat server: Create a private channel for the user and the moderator team. This can be done with the bot's `private` command.

- The user is guaranteed some privacy.
- Public chat is no longer disrupted.
- Other team members can pitch in, should you feel uncomfortable dealing with the situation yourself.

Helpful URLs:

http://suicide.org/international-suicide-hotlines.html

## A note on free speech

Sometimes people will defend something offensive or incendiary that they said as "free speech."

This XKCD comic summarizes perfectly most communities' thoughts on free speech. So if someone defends something in the name of "free speech", feel free to send it to them.

> <div align="center"><img src='https://imgs.xkcd.com/comics/free_speech.png' width="400" height="400" /></div>

Reviewer 2: ```suggestion
<div align="center"><img src='images/github/xkcd-free-speech.png' width="400" height="400" /></div>
```
This should work for the link.

As a matter of policy, you should avoid talking privately about these matters. Should the situation reflect back to freeCodeCamp, we want to have the conversation(s) on record. Make it clear that we are not medical professionals and that you encourage the user to find professional help.

As difficult as it sometimes can be, avoid giving any tips or advice and rather point the user in the direction of seeking professional help!

If this happens on our chat server: Create a private channel for the user and the moderator team. This can be done with the bot's `private` command.

- The user is guaranteed some privacy.
- Public chat is no longer disrupted.
- Other team members can pitch in, should you feel uncomfortable dealing with the situation yourself.

Helpful URLs:

http://suicide.org/international-suicide-hotlines.html

> ## A note on free speech

Reviewer 2: ```suggestion
## A Note on Free Speech
```

Sometimes people will defend something offensive or incendiary that they said as "free speech."

This XKCD comic summarizes perfectly most communities' thoughts on free speech. So if someone defends something in the name of "free speech", feel free to send it to them.

> <div align="center"><img src='https://imgs.xkcd.com/comics/free_speech.png' width="400" height="400" /></div>

Reviewer: hosting it ourself means we don't ping xkcd servers each time the image is loaded. I wouldn't change this.

Author: Yes, I understand, I will definitely need a new image link to update this as the prior link ends up with this:
![image](https://user-images.githubusercontent.com/97711937/181085803-f1f3e9ac-a1b3-4f6f-ba5f-4bc6c1e19fcc.png)

Reviewer: images are saved in docs/images for the docs, it can be added there

Author: Should I be creating a separate pull request to add an image like that to `freeCodeCamp/docs/images`?

Reviewer: as you are changing the link here it should be here

Author: Unfortunately I don't see the image of XKCD in `freeCodeCamp/docs/images`.

Reviewer: Yeah, it has to be added.

Author: So I would have to make a pull request with the file to that subdirectory, correct?

Reviewer: You can add it in this PR

Thanks for reading this, and thanks for helping the developer community!

> ## Reply templates

Reviewer 2: ```suggestion
## Reply Templates
```

These are some of the standard reply templates that you may use while reviewing pull requests and triaging issues/pull requests.

> You can make your own saved replies with GitHub's built-in [saved replies](https://github.com/settings/replies/) feature or use the ones below.

> ### Thank you

Reviewer 2: ```suggestion
### Thank You
```

We are happy to accept these changes and look forward to future contributions. 📝
```

> ### Build error

Reviewer 2: ```suggestion
### Build Error
```

Feel free to reference the [contributing guidelines](how-to-work-on-coding-challenges.md#testing-challenges) for instructions on running the CI build locally. ✅
```

> ### Syncing fork

Reviewer 2: ```suggestion
### Syncing Fork
```

---

Feel free to reference the ["Syncing a fork"](https://help.github.com/articles/syncing-a-fork/) article on GitHub for more insight on how to keep your fork up-to-date with the upstream repository. 🔄
````

> ### Merge conflicts

Reviewer 2: ```suggestion
### Merge Conflicts
```

---

If you have any questions, feel free to ask questions on the ["Contributors" category on our forum](https://forum.freecodecamp.org/c/contributors) or [the contributors chat room](https://discord.gg/PRyKn3Vbay).
```

> ### Closing invalid pull requests

Reviewer 2: ```suggestion
### Closing Invalid Pull Requests
```

We are closing this pull request. Please suggest links and other details to add the challenge's corresponding guide post through [a forum topic](https://forum.freecodecamp.org/new-topic?category=Contributors&title=&body=**What%20is%20your%20hint%20or%20solution%20suggestion%3F**%0A%0A%0A%0A%0A**Challenge%3A**%0A%0A%0A**Link%20to%20the%20challenge%3A**) instead.

If you think we're wrong in closing this issue, please request for it to be reopened and add further clarification. Thank you and happy coding.
```

> ### Closing invalid issues

Reviewer 2: ```suggestion
### Closing Invalid Issues
```

Thank you and happy coding.
```

> When an issue is duplicate of an earlier issue.

```markdown
Thank you for reporting this issue.

> This is a standard message notifying you that this issue appears to be very similar to issue <#number>, so we are closing it as a duplicate.

Reviewer: ```suggestion
This is a standard message notifying you that this issue appears to be very similar to issue #XXXXX, so we are closing it as a duplicate.
```

as this is a prewritten message, having it as `#XXXXX` means we have to change less characters, as that's the format used on GitHub to reference issues and PRs, like `#47039` becoming #47039

If you think we're wrong in closing this issue, please request for it to be reopened and add further clarification. Thank you and happy coding.
```

> ### First-timer only issues

Reviewer 2: ```suggestion
### `first timer only` Issues
```

Let's make this a code tag, and match the actual label.
